cmd/kaspawallet/sweep.go

```diff
@@ -153,43 +146,38 @@ func createSplitTransactionsWithSchnorrPrivteKey(
 
 	var splitTransactions []*externalapi.DomainTransaction
 
-	// I need to add extra mass to transaction, txmasscalculater seems to undercalculate, by around this amount.
+	// Add extra mass to transaction, to account for future signatures.
 	extraMass := uint64(7000)
 	massCalculater := txmass.NewCalculator(params.MassPerTxByte, params.MassPerScriptPubKeyByte, params.MassPerSigOp)
 
-	//for refernece this keeps track in respect to dummyWindow[0], not dummyWindow[1]
 	currentIdxOfSplit := 0
 
 	totalAmount := uint64(0)
 
 	totalSplitAmount := uint64(0)
 
-	ScriptPublicKey, err := txscript.PayToAddrScript(toAddress)
+	scriptPublicKey, err := txscript.PayToAddrScript(toAddress)
 	if err != nil {
 		return nil, err
 	}
 
-	dummyTransactionWindow := make([]*externalapi.DomainTransaction, 2)
-	//[0] is the last build that didn't violate mass, that can be added as a split
-	dummyTransactionWindow[0] = newDummyTransaction()
-	//[1] represents the tested unsigned transaction
-	dummyTransactionWindow[1] = newDummyTransaction()
+	lastValidTx := newDummyTransaction()
+	currentTx := newDummyTransaction() //i.e. the tested tx
 
 	//loop through utxos commit segments that don't violate max mass
 	for i, currentUTXO := range selectedUTXOs {
 
 		if currentIdxOfSplit == 0 {
```

Comment:
I think this variable is actually doing nothing. On the first iteration `lastValidTx` and `currentTx` are actually fresh and empty, and you reset them every time you set `currentIdxOfSplit` to `0`

Reply:
`currentIdxOfSplit`? this is needed to reset `lastValidTx` and `currentTx`, for a new split transaction. I agree it doen't do anything during the first iteration, so if you want I can do an extra check with `&& i != 0` before entering the scope. But it is needed to reset the Txs after each split, excluding the first.